search/query_parser.go

```diff
@@ -0,0 +1,90 @@
+package search
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// DID is not currently being used presently
+type DidHandle struct {
```

Comment:
This slightly futureproofs things when the user DID gets added into the `posts` index mapping, so we can do a lookup from handle to DID and hold them in a logical struct instead of two separate string fields.